### PR TITLE
yx - Fixed Create Announcement

### LINKED ISSUE #24: Create Create Announcements Page
EPIC: Frontend for Announcements on Play Page 
Issue F2
Create "Create Announcements Page"

### LINKED ISSUE #32: BUG: Fix Create Announcement
### Summary
Create announcement is requiring commonsId as an input while it should be implicitly taken.

### Expected Behavior 
Should post normally without errors and should retrieve commonsId implicitly.

### Current/Observed Behavior
image

### PR DESCRIPTION
**Pull request by yixuan-wong**
## Overview
Issues:
- Created the Create Announcements Page
- Fixed POST for create announcements to no longer require commonsId as an input and instead be implicitly set. 
- Fixed difference in formatting for date between frontend and backend that was causing issues. 
- Start and end date is now in simple date format YYYY-mm-dd instead of ISO format.
- Implemented start date now defaults to current date if a start date is not set.

## Screenshots
Announcements start date defaults to current date:
image

Announcements with no end date:
image

Announcements with both start and end date:
image

Only announcements is required (no end date will default to null):
image

Verification: 
CommonsId should be 3:
image

## Validation
1. Click on "Admins" in the top bar and click on "List Commons".
2. Scroll to the far right on the table and click on "Announcements". (Note the commonsId)
3. Click on "Create Announcement".
4. Create an announcement.
5. Click on Swagger in the top bar. 
6. Scroll down to "Announcements" and click on "Get all announcements"
7. Enter in the commonsId of the commons tested earlier and verify announcements have been posted.

## Future Possibilities
Maybe add in a feature to add time for announcements as well. 

## Tests
- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Development Deployment Link
https://happycows-dev-yixuan-wong.dokku-12.cs.ucsb.edu 

## Linked Issues
Closes #24 #32 #48

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "nock": "^13.2.4",
     "react": "^17.0.1",
     "react-bootstrap": "^2.0.0-alpha.0",
+    "react-datepicker": "^7.5.0",
```
Comment (pconrad) [inline frontend/package.json]: Why are we adding this as a dependency when I don't see it used anywhere in the code?

I'm not opposed to adding this, but if you add it as a dependency, you need to actually use it somewhere in the PR.  So either use it, or lose it.

Comment (pconrad) [reply frontend/package.json]: Oh, and if you do decide to lose it, make sure that package-lock.json gets restored as well: 

See: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

**Review by pconrad — CHANGES_REQUESTED** [review]
Please either use react-datepicker somewhere in your code,  or remove it from the PR.

We try to avoid taking on needless dependencies.

image

File: frontend/package.json (hunk 1 above)
Comment (yixuan-wong) [reply frontend/package.json]: I have removed react-datepicker from both package.json and package-lock.json.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please see the code review feedback on #28 ... the same issues apply here.

**Review by yungongWang — DISMISSED** [review]
Great job! All the function in announcement creating page works well. Also, the Swagger api GET can also work. I can see that all the Stryker exceptions are copied from other example files, so I have approved it. Thank you for your time and effort.

**Pull request merged**